### DISCUSSION TITLE
Does HW1 Parallel support high impedence pins for 3.3V systems?

### DISCUSSION THREAD

**Discussion by V-FEXrt** [darthcloud/BlueRetro#627] [Q&A] [answered]
Hi everyone! 

I'm building HW1 (DIY) following https://github.com/darthcloud/BlueRetro/wiki/BlueRetro-Cables-Build-Instructions#parallel-1p-5v-12-buttons-adapter-cable but the target system is 3.3v matching the esp32. I want to be able to direct connect the pins to the system but it seems that the firmware is actively driving the pins high instead of setting them as inputs/setting high impedance. 

It seems that this is supported for HW2 per https://github.com/darthcloud/BlueRetro/wiki/BlueRetro-HW2-Internal-Install-Specification#3v3-level-bus and it would be nice to use this in HW1 as well.

I flashed firmware `BlueRetro_parallel_1p_3v3.bin` @ `v1.8.3_hw1`  if that is useful info.

Thanks!

**Comment by darthcloud** [comment, marked as answer]
The 3v3 parallel FWs for both HW1 & HW2 are configuring the pins as open drain.

But the weak pull-up is enable on the pins, so that might be what you see.

**Comment by V-FEXrt** [comment]
I see. Thanks! That makes sense. 

I am seeing some really strange behavior but the problem is most likely that I have two different 3v3 sources (battery 3v3 and the regulated 3v3 from the esp32 via VUSB). But you are right, I thought it was actively driving high because I saw the values pulled to 3v3 when I probed just the esp32
